scripts/local_k8s/unittests/test_cluster_creation.py

```
@@ -0,0 +1,201 @@
+class TestClusterCreation:
+    """Provider command-contract tests for preview cluster creation."""
+
+    def test_k3d_cluster_creation_keeps_loopback_load_balancer(
+        self,
+        monkeypatch: pytest.MonkeyPatch,
+        preview_config: PreviewConfig,
+    ) -> None:
+        """Verify the default k3d path keeps its existing port mapping contract."""
+
+        def respond(
+            command: str, args: list[str], _input_text: str | None
```

Ruff (0.15.20)

[warning] Unused function argument: `command`

(ARG001)